compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/repositories/FileSystemRepository.java

```diff
@@ -172,9 +173,54 @@ public FileSystemRepository(Environment environment, Path cacheDirectory, String
         } catch (IOException e) {
             throw new RuntimeException("Error while accessing Distribution cache: " + e.getMessage());
         }
+
+        List<Path> incompatibleVersions = new ArrayList<>();
+
+        if (!versions.isEmpty()) {
+            for (Path ver : versions) {
+                Path pkgJsonPath = bala.resolve(org.value()).resolve(name.value()).resolve(ver.toString())
+                        .resolve("any").resolve(ProjectConstants.PACKAGE_JSON);
+
+                if (Files.exists(pkgJsonPath)) {
+                    String packageVer = BalaFiles.readPkgJson(pkgJsonPath).getBallerinaVersion()
+                            .replace("sl", "");
+                    String packVer = RepoUtils.getBallerinaShortVersion().replace("sl", "");
```

Comment:
we can remove `slbeta` when trimming the version string and compare the remaining number

Reply:
Removed trimming